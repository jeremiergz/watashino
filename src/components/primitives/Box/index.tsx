import React, { TimeHTMLAttributes } from 'react';
import styled, { StyledComponentProps } from 'styled-components';
import {
    border,
    BorderProps,
    color,
    ColorProps,
    compose,
    flexbox,
    FlexboxProps,
    layout,
    LayoutProps,
    space,
    SpaceProps,
    typography,
    TypographyProps,
    zIndex,
    ZIndexProps,
} from 'styled-system';
import { Theme, ThemeProps } from '../../../theme';

type BoxProps = StyledComponentProps<
    'div',
    Theme,
    BorderProps &
        ColorProps &
        FlexboxProps &
        LayoutProps &
        SpaceProps &
        Omit<TypographyProps, 'fontWeight'> &
        ZIndexProps &
        TimeHTMLAttributes<unknown> &
        ThemeProps,
    never
> & {
    as?: keyof JSX.IntrinsicElements | React.ComponentType<unknown>;
};

const Box = styled.div<BoxProps>`
    ${compose(
        border,
        color,
        flexbox,
        layout,
        space,
        typography,
        zIndex,
    )}
`;

Box.displayName = 'Box';

export { BoxProps };
export default Box;
