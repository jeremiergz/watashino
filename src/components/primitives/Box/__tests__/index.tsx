import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Box from '..';

describe('components/primitives/Box Test Suite', () => {
  it('matches snapshot', () => {
    const jsx = <Box>Box Component</Box>;
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
