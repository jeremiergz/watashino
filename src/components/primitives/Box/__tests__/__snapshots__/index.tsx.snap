// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/primitives/Box Test Suite matches snapshot 1`] = `
<div
  className="sc-AxjAm hyNcgC"
>
  Box Component
</div>
`;
